uefi/src/proto/network/http.rs

```diff
@@ -0,0 +1,370 @@
+#![cfg(target_os = "uefi")]
+
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! HTTP Protocol
+
+extern crate alloc;
+use alloc::string::String;
+use alloc::vec;
+use alloc::vec::Vec;
+use core::ffi::{c_char, c_void, CStr};
+use core::ptr;
+use log::debug;
+
+use uefi::boot::ScopedProtocol;
+use uefi::prelude::*;
+use uefi::proto::unsafe_protocol;
+use uefi_raw::protocol::driver::ServiceBindingProtocol;
+use uefi_raw::protocol::network::http::{
+    HttpAccessPoint, HttpConfigData, HttpHeader, HttpMessage, HttpMethod, HttpProtocol,
+    HttpRequestData, HttpResponseData, HttpStatusCode, HttpToken, HttpV4AccessPoint, HttpVersion,
+};
+
+/// HTTP Protocol
+#[derive(Debug)]
+#[unsafe_protocol(HttpProtocol::GUID)]
+pub struct Http(HttpProtocol);
+
+impl Http {
```

Comment:
isn't this missing `mode_data()` (`GetModeData`)?

Reply:
I have not used and not tested that one.  `cancel` is missing too btw.  Will look into that.